### DISCUSSION TITLE
GitHub inserts periods in uploaded file name with spaces

### DISCUSSION THREAD

**Discussion by rrirower** [community/community#60449] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I uploaded a release file with spaces in the name.  When it was uploaded, GitHub replaced the spaces with periods ".".  Is there any way to rename the file to use spaces?

**Comment by anhtran917** [comment]
Hi, @rrirower 
Yes, you can rename the file to use spaces instead of periods in GitHub. 

Go to the repository where the file is located on GitHub and navigate to the file you want to rename.
Then click on the file to view its contents.
And at the top-right corner of the file view, you can see a pencil icon. Click on it to edit the file.
Modify the file name, replacing the periods with spaces.
Then click "Commit changes."

After that you can change the file name with space.

**Reply by rrirower** [reply]
Thanks for your reply. Is that possible if the file I want to rename was uploaded as part of a release?

**Reply by anhtran917** [reply]
Oh, well sorry. In that case, you need to delete current file and upload again it with space in name.
Did you ever do it?

**Reply by rrirower** [reply]
Yes, I have. The original file name contains spaces and is an "installer" executable program. I upload it and GitHub replaces the spaces with periods. This is a problem for me because I have some code that looks for the file without spaces and does not find it because GitHub has inserted the periods.

**Reply by anhtran917** [reply]
I understood. It seems that directly renaming the file within GitHub to use spaces instead of periods is not possible.
The automatic replacement of spaces with periods is a built-in behavior of GitHub's file upload process.
So I think it's better to modify code to find the file with periods.

**Reply by rrirower** [reply]
I was afraid of that. Thank you for your advice.

**Reply by anhtran917** [reply]
Sorry for that, but I think it is not difficult to replace all the part of file name in the project.

**Comment by i2van** [comment]
Renaming uploaded file in `releases/edit` still changes spaces to dots. What is the justification behind it?

**Reply by ajilani** [reply]
I would also like to know this. It causes minor issues in our workflow. We could work around the issue, but I don't understand why GitHub is changing the file name in the first place.
